cmd/cosign/cli/sign.go

```diff
@@ -132,11 +138,28 @@ func SignCmd(ctx context.Context, keyPath string,
 		if err != nil {
 			return errors.Wrap(err, "getting public key")
 		}
-	} else {
+	case keyPath != "":
 		signature, publicKey, err = sign(ctx, get, keyPath, payload, pf)
-	}
-	if err != nil {
-		return errors.Wrap(err, "signing")
+		if err != nil {
+			return errors.Wrap(err, "signing payload")
+		}
+	default: // Keyless!
+		fmt.Fprintln(os.Stderr, "Generating ephemeral keys...")
```

Comment:
do you have a  logging framework yet?

Reply:
No logging yet (it should all be debug when we add it). Right now this is all output directly intended for users. I usually think of 3 kinds of output in CLI tools:

- Results (parseable, to stdout)
- Messages (text, to stderr)
- Logging (debug, hidden by default, used to diagnose problems)

Reply:
fair point! the user CLI is better without standard logging